lib/SodiumCrypto.js

```diff
@@ -590,4 +590,20 @@ export default class SodiumCrypto extends Crypto {
     // Returns a File object
     return decryptedBlob
   }
+
+  /**
+   * Helper function for tsv1 which wraps sodium's crypto_generichash function, uses BLAKE2b. https://libsodium.gitbook.io/doc/hashing/generic_hashing
+   */
+  genericHash(rawString) {
+    return sodium.crypto_generichash(sodium.crypto_generichash_BYTES, rawString)
+  }
+
+  /**
+   * Helper function for tsv1 which wraps sodium's cyrpto_sign_detached fuction, https://libsodium.gitbook.io/doc/public-key_cryptography/public-key_signatures.
+   * Creates signature without attaching a copy of the original message.
+   *
+   */
+  signDetached(stringToSign, privateKey) {
```

Comment:
sign detached is used for https://github.com/tozny/browser-sodium-sdk/blob/master/lib/SodiumCrypto.js#L199 as well. That return is a base64 encoded string. The two methods are slightly different I suppose (this is not a 'document' with a `serialize()` method). Does it make sense to consolidate, or at least relate the two methods where one calls the other and abstract the return as a b64url encoded string instead of raw bytes to keep the overall abstraction level at a similar level?